data-prepper-plugins/drop-events-processor/src/main/java/com/amazon/dataprepper/plugins/processor/drop/DropEventsProcessor.java

```diff
@@ -5,31 +5,48 @@
 
 package com.amazon.dataprepper.plugins.processor.drop;
 
+import com.amazon.dataprepper.metrics.PluginMetrics;
 import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
 import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
-import com.amazon.dataprepper.model.configuration.PluginSetting;
 import com.amazon.dataprepper.model.event.Event;
-import com.amazon.dataprepper.model.record.Record;
 import com.amazon.dataprepper.model.processor.AbstractProcessor;
 import com.amazon.dataprepper.model.processor.Processor;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.amazon.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+
 import java.util.Collection;
 import java.util.Collections;
+import java.util.stream.Collectors;
 
-@DataPrepperPlugin(name = "drop_events", pluginType = Processor.class)
+@DataPrepperPlugin(name = "drop_events", pluginType = Processor.class, pluginConfigurationType = DropEventProcessorConfig.class)
 public class DropEventsProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
 
-    private static final Logger LOG = LoggerFactory.getLogger(DropEventsProcessor.class);
+    private final DropEventsWhenCondition whenCondition;
 
     @DataPrepperPluginConstructor
-    public DropEventsProcessor(final PluginSetting pluginSetting) {
-        super(pluginSetting);
+    public DropEventsProcessor(
+            final PluginMetrics pluginMetrics,
+            final DropEventProcessorConfig dropEventProcessorConfig,
+            final ExpressionEvaluator<Boolean> expressionEvaluator
+    ) {
+        super(pluginMetrics);
+
+        whenCondition = new DropEventsWhenCondition.Builder()
+                .withDropEventsProcessorConfig(dropEventProcessorConfig)
+                .withExpressionEvaluator(expressionEvaluator)
```

Comment:
I am concerned about our ConditionalExpressionEvaluator threadsafety. If it is not threadsafe, we might run into issue as DI only creates a singleton shared by all workers accessing the processor. 

But if `ParseTreeEvaluatorListener` is the only component that is stateful, we should be safe as it is now created on demand with event evaluation.